src/provider/PocketProvider.js

```diff
@@ -0,0 +1,51 @@
+import {Provider} from "./Provider";
+import {Pocket, PocketAAT, typeGuard} from "@pokt-network/pocket-js";
+import {RpcError} from "@pokt-network/pocket-js/lib/src/rpc";
+
+export class PocketProvider extends Provider {
+
+    constructor(token, baseServer, port, headers = {}, pocketPrivateKey, pocketPublicKey, pocketAddress, pocketPassphrase) {
+        super(token, baseServer, port, headers);
+
+        this.pocket = new Pocket([new URL(this.address)])
+
+        this.pocket.keybase.importAccount(Buffer.from(pocketPrivateKey, 'hex'), pocketPassphrase).then(() => {
+            this.pocket.keybase.unlockAccount(pocketAddress, pocketPassphrase, 0).then(() => {
+                (async () => {
+                    try {
+                        this.pocketAAT = await PocketAAT.from('0.0.1', pocketPublicKey, pocketPublicKey, pocketPrivateKey)
```

Comment:
I think it is better form to create the AAT outside of the provider and then pass it in -- so like in the ICON demo file, I leave it to the dev to create the AAT and then instantiate the provider.

That leaves the way they create the AAT flexible rather than forcing them to hardcode private keys

Reply:
Understood